src/storage/storage_util.cpp

```diff
@@ -141,6 +141,42 @@ std::vector<uint16_t> StorageUtil::ComputeBaseAttributeOffsets(const std::vector
   return offsets;
 }
 
+uint8_t StorageUtil::AttrSizeFromBoundaries(const std::vector<uint16_t> &boundaries, const uint16_t col_idx) {
+  TERRIER_ASSERT(boundaries.size() == NUM_ATTR_BOUNDARIES,
+                 "Boudaries vector size should equal to number of boundaries");
+  uint8_t shift;
+  for (shift = 0; shift < NUM_ATTR_BOUNDARIES; shift++) {
+    if (col_idx < boundaries[shift]) break;
+  }
+  TERRIER_ASSERT(shift <= NUM_ATTR_BOUNDARIES, "Out-of-bounds attribute size");
+  TERRIER_ASSERT(shift >= 0, "Out-of-bounds attribute size");
+  return static_cast<uint8_t>(16U >> shift);
```

Comment:
The code above deserves a comment describing what's going on since it isn't entirely obvious bit arithmetic.  It's OK if it is just duplicating the explanation from elsewhere, but it should be here as well

Reply:
Done.